src/lib/Form/Type/Content/Draft/ContentCreateType.php

```diff
@@ -119,6 +120,21 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'content_draft_create_type.create',
                 ]
             );
+
+        $builder->addEventListener(
+            FormEvents::PRE_SUBMIT,
+            function (PreSubmitEvent $event): void {
+                $location = $this->locationService->loadLocation(
+                    (int)$event->getData()['parent_location']
+                );
+                $form = $event->getForm();
+                $opts = $form->get('content_type')->getConfig()->getOptions();
+                $opts['choice_loader'] = $this->contentCreateContentTypeChoiceLoader
+                    ->setTargetLocation($location);
+                $form->remove('content_type');
+                $form->add('content_type', ContentTypeChoiceType::class, $opts);
```

Comment:
I'm not a fan of removing and adding the field back in almost all cases.

Can addition of this field be moved into the event, so it is created here? What about the `PRE_SET_DATA` event?

Isn't the language choice loader also affected by this, since it is also dependent (somewhat) on the selected parent location?